Add VPC Flow documentation for the GCP module (#3911)
wazuh/wazuh-documentation: source/gcp/supported-services/vpc_flow.rst

.. Copyright (C) 2021 Wazuh, Inc.

.. _gcp_vpc_flow:

VPC Flow logs
=============

> `VPC Flow Logs <https://cloud.google.com/vpc/docs/flow-logs>`__ records a sample of network flows sent from and received by VM instances, including instances used as Google Kubernetes Engine nodes. Flow logs are aggregated by connection from Compute Engine VMs and exported in real time.

Reviewer: ```suggestion
`VPC Flow Logs <https://cloud.google.com/vpc/docs/flow-logs>`__ record a sample of network flows sent from and received by VM instances, including instances used as Google Kubernetes Engine nodes. Flow logs are aggregated by connection from Compute Engine VMs and exported in real time.
```

Author: In this case "VPC Flow Logs" is the whole name of the service, being "Logs" part of its name, so using the singular is the right choice here.

> `VPC Flow Logs <https://cloud.google.com/vpc/docs/flow-logs>`__ records a sample of network flows sent from and received by VM instances, including instances used as Google Kubernetes Engine nodes. Flow logs are aggregated by connection from Compute Engine VMs and exported in real time.

Reviewer 2: ```suggestion
`VPC Flow Logs <https://cloud.google.com/vpc/docs/flow-logs>`__ records a sample of network flows sent from and received by VM instances, including instances used as Google Kubernetes Engine nodes. Flow logs are aggregated by the connection from Compute Engine VMs and exported in real time.
```

> .. note:: To be able to send VPC Flow logs to Wazuh Cloud Logging must be configured to export these logs to Pub/Sub first. More information about how Pub/Sub works and how to configure it :ref:`here <pubsub>`.

Reviewer: ```suggestion
.. note:: To be able to send VPC Flow logs to Wazuh, Cloud Logging must be configured to export these logs to Pub/Sub first. More information about how Pub/Sub works and how to configure it :ref:`here <pubsub>`.
```

Enabling VPC Flow Logs
----------------------

> VPC Flow Logs can be enabled in the VPC networks page in the Google Cloud Console. It can be enabled both for new and existing subnets. Follow its `official documentation <https://cloud.google.com/vpc/docs/using-flow-logs#enabling-vpc-flow-logs>`__ for the most up-to-date instructions on how to enable this feature.

Reviewer: ```suggestion
VPC Flow Logs can be enabled in the VPC networks page in the Google Cloud Console. They can be enabled for both new and existing subnets. Follow its `official documentation <https://cloud.google.com/vpc/docs/using-flow-logs#enabling-vpc-flow-logs>`__ for the most up-to-date instructions on how to enable this feature.
```

Exporting VPC Flow logs to Pub/Sub
----------------------------------

> In the :ref:`Pub/Sub <pubsub>` page we explained how to set up a sink to logs to a Pub/Sub topic. However, this would export every single log available, and not only the VPC Flow logs. It is possible to configure the Sink to export VPC Flow logs only to a topic, ignoring logs coming from other services, by adding a filtering condition to the Sink. To do so, follow the same instructions as explained in the :ref:`Pub/Sub <pubsub>` section but adding the following filter in **Step 3 - Choose logs to include in sink**:

Reviewer: ```suggestion
In the :ref:`Pub/Sub <pubsub>` page, it was explained how to set up a sink to logs for a Pub/Sub topic. However, this would export every single log available, and not only the VPC Flow logs. It is possible to configure the sink to export VPC Flow logs only to a topic, ignoring logs coming from other services, by adding a filtering condition to the sink. To do so, follow the same instructions as explained in the :ref:`Pub/Sub <pubsub>` section but adding the following filter in **Step 3 - Choose logs to include in sink**:
```

> In the :ref:`Pub/Sub <pubsub>` page, it was explained how to set up a sink to logs for a Pub/Sub topic. However, this would export every single log available, and not only the VPC Flow logs. It is possible to configure the sink to export VPC Flow logs only to a topic, ignoring logs coming from other services, by adding a filtering condition to the sink. To do so, follow the same instructions as explained in the :ref:`Pub/Sub <pubsub>` section but adding the following filter in **Step 3 - Choose logs to include in sink**:

Reviewer 2: ```suggestion
In the :ref:`Pub/Sub <pubsub>` page, it was explained how to set up a sink to logs for a Pub/Sub topic. However, this would export every single log available, and not only the VPC Flow logs. It is possible to configure the sink to export VPC Flow logs only to a topic, ignoring logs coming from other services, by adding a filtering condition to the sink. To do so, follow the same instructions as explained in the :ref:`Pub/Sub <pubsub>` section but adding the following filter in **Step 3 - Choose logs to include in the sink**:
```

Author: `Choose logs to include in sink` is the name of the Step 3 as appears in the screenshot below that paragraph. We should mantain the name as is.

Reviewer 2: Agree
